src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,615 @@
+        self.signal_map[f"measure{q}"] = self.device_setup.logical_signal_groups[f"q{q}"].logical_signals[
+            "measure_line"
+        ]
+        self.calibration[f"/logical_signal_groups/q{q}/measure_line"] = lo.SignalCalibration(
+            oscillator=lo.Oscillator(
+                frequency=intermediate_frequency,
+                modulation_type=modulation_type,
+            ),
+            local_oscillator=lo.Oscillator(
+                frequency=int(qubit.readout.local_oscillator.frequency),
+            ),
+            range=qubit.readout.power_range,
+            port_delay=None,
+            delay_signal=0,
+        )
+
+        self.signal_map[f"acquire{q}"] = self.device_setup.logical_signal_groups[f"q{q}"].logical_signals[
+            "acquire_line"
+        ]
+        self.calibration[f"/logical_signal_groups/q{q}/acquire_line"] = lo.SignalCalibration(
+            oscillator=lo.Oscillator(
+                frequency=intermediate_frequency,
+                modulation_type=lo.ModulationType.SOFTWARE,
+            ),
+            local_oscillator=lo.Oscillator(
+                frequency=int(qubit.readout.local_oscillator.frequency),
+            ),
+            range=qubit.feedback.power_range,
+            port_delay=None,
+            delay_signal=0,
+        )
+
+    def register_drive_line(self, qubit, intermediate_frequency):
+        """Registers qubit drive line to calibration and signal map."""
+        q = qubit.name
+        self.signal_map[f"drive{q}"] = self.device_setup.logical_signal_groups[f"q{q}"].logical_signals["drive_line"]
+        self.calibration[f"/logical_signal_groups/q{q}/drive_line"] = lo.SignalCalibration(
+            oscillator=lo.Oscillator(
+                frequency=intermediate_frequency,
+                modulation_type=lo.ModulationType.HARDWARE,
+            ),
+            local_oscillator=lo.Oscillator(
+                frequency=int(qubit.drive.local_oscillator.frequency),
+            ),
+            range=qubit.drive.power_range,
+            port_delay=None,
+            delay_signal=0,
+        )
+
+    def register_flux_line(self, qubit):
+        """Registers qubit flux line to calibration and signal map."""
+        q = qubit.name
+        self.signal_map[f"flux{q}"] = self.device_setup.logical_signal_groups[f"q{q}"].logical_signals["flux_line"]
+        self.calibration[f"/logical_signal_groups/q{q}/flux_line"] = lo.SignalCalibration(
+            range=qubit.flux.power_range, port_delay=None, delay_signal=0, voltage_offset=qubit.flux.offset
+        )
+
+    def run_exp(self):
+        # Compiler settings required for active reset and multiplex.
+        compiler_settings = {
+            "SHFSG_FORCE_COMMAND_TABLE": True,
+            "SHFSG_MIN_PLAYWAVE_HINT": 32,
+            "SHFSG_MIN_PLAYZERO_HINT": 32,
+        }
+
+        self.exp = self.session.compile(self.experiment, compiler_settings=compiler_settings)
+        self.results = self.session.run(self.exp)
```

Comment:
I think it would help readability to rename `self.exp` to `self.compiled_experiment`.

In fact you can easily remove all these attributes (`self.experiment`, `self.exp`, `self.results`) by just returning them and making the play sequence in `play` and `sweep` look like
```py
self.sequence_zh(sequence, qubits, sweepers)
experiment = self.create_exp(qubits, nshots, relaxation_time)
results = self.run_exp(experiment)
```
however I understand you prefer to keep them for debugging reasons (eg. easy access in an interactive session).

Reply:
At least for now I would prefer to keep them, they are helpful